### PR TITLE
Add adapter for ADAO

### PR DESCRIPTION
**Pull request by githubDo111**
##### Twitter Link:
https://twitter.com/ADAO_Official

##### Website Link:
https://www.adaotop.com/

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
https://www.adaotop.com/images/logo.webp

##### Current TVL:
astar-staking             46.38 k
astar-treasury            0.15911129999980486
astar                     46.38 k
staking                   46.38 k
treasury                  0.15911129999980486

total                    46.38 k 

##### Chain:
Astar

##### Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)
N/A

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
N/A

##### Short Description (to be shown on DefiLlama):
A-DAO will be based on dApp staking of Astar Network. Users will get some of the developer rewards while participating and gaining basic rewards. At present, A-DAO divides the developer rewards into: Revenue Reward, On-chain Treasury, Incubation Fund, any rewards of which can be adjusted by DAO governance.

##### Token address and ticker if any:
ibASTR Token:  0x3BFcAE71e7d5ebC1e18313CeCEbCaD8239aA386c

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Defi

##### Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):
N/A

##### forkedFrom (Does your project originate from another project):
N/A

##### methodology (what is being counted as tvl, how is tvl being calculated):
Amount of ibASTR

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/adao exports TVL: 
        
 
 
    astar-staking             47.83 k
    astar                     47.83 k
    astar-treasury            0.16436465999979843
    staking                   47.83 k
    treasury                  0.16436465999979843
    
    total                    47.83 k

File: projects/adao/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  let balance = await dappContract.methods.totalSupply().call();
+  balance = web3.utils.fromWei(balance, "ether");
+  balances[`astar`] = Number(balance);
+  return balances;
+}
+module.exports = {
+  methodology:
+    "A-DAO will be based on dApp staking of Astar Network. Users will get some of the developer rewards while participating and gaining basic rewards. At present, A-DAO divides the developer rewards into: Revenue Reward, On-chain Treasury, Incubation Fund, any rewards of which can be adjusted by DAO governance.",
+  astar: {
+    tvl: staking,
```
Comment (waynebruce0x) [inline projects/adao/index.js:31]: can you change this to:
tvl: () => ({})

Comment (githubDo111) [reply projects/adao/index.js]: already change, thanks

**Comment by llamatester** [comment]
The adapter at projects/adao exports TVL: 
        
 
 
    astar-treasury            0.2768945999998106
    astar-staking             44.80 k
    astar                     44.80 k
    treasury                  0.2768945999998106
    staking                   44.80 k
    
    total                    44.80 k

**Pull request merged**